### PR TITLE
Feature/ecr repositories page

### PR DESCRIPTION
**Pull request by andreasfrisch**
closes dfds/cloudplatform/issues/2006

# Additional Review Notes
I strongly believe we should just have a single naming column, namely the repository name.
This current additional name feels wrong to work with and does not fit the domain.

Simply removing the last PR is 99% of what is required for not using the name for anything.

### CODE REVIEW

File: src/src/pages/ecr/index.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            surface="main"
+            size="xl"
+            reverse={true}
+          >
+            <CardTitle largeTitle>Information</CardTitle>
+            <CardContent>
+              <p>
+                This is a long text about ECR Repositories. We have no idea what
+                goes here and it is just a placeholder. However, if it is too short
+                then the layout will be all wrong. Long live Lorem Ipsum and all that.
```
Comment (kralle333) [inline src/src/pages/ecr/index.js:128]: Should we have a more helpful message here?

File: src/src/pages/ecr/NewRepositoryDialog.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            />
+
+          <TextField
+            label="Repository Name"
+            placeholder="Enter a name for the new repository"
+            required
+            value={formData.repositoryName}
+            onChange={changeRepositoryName}
+            errorMessage={repositoryNameErrorMessage}
+          />
```
Comment (kralle333) [inline src/src/pages/ecr/NewRepositoryDialog.js:107]: Would be nice with some help text to show that the format is normally `some-team/repo-name`. 
Some examples we have:
```
passengeraccounting/adjuapproval
selfservice/ascraper
uiam/uiam-dev-ui
ded/p-team/p-app/dbmigrations
apt/fe
passengerships/machinelearning
dfds-baseimages/aspnet-runtime-5.0
logistics-invoicing/logistics/invoicing-service
```

File: src/src/pages/ecr/NewRepositoryDialog.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  
+  let repositoryNameErrorMessage = "";
+  if (formData.repositoryName.length > 0 && !isRepositoryNameValid) {
+    repositoryNameErrorMessage =
+      'Allowed characters are a-z, 0-9, "-", "_", and "/". Repositories must not start or end with "_", "-", or "/". Do not use more than one of "-", "_", or "/" in a row.';
+  }
+  
+  if (formData.repositoryName.length > 150) {
+    repositoryNameErrorMessage = "Please consider a shorter name.";
+  }
```
Comment (kralle333) [inline src/src/pages/ecr/NewRepositoryDialog.js:64]: does it match the aws repo restrictions?

**Review by kralle333 — CHANGES_REQUESTED** [review]
Very nice!
I have some smaller comments related to ui/ux, but nothing codewise ;)

File: src/src/pages/ecr/NewRepositoryDialog.js (hunk 2 above)
Comment (andreasfrisch) [reply src/src/pages/ecr/NewRepositoryDialog.js:107]: Do we want to figure out a way to enforce this.
The only rule from AWS is 'no spaces or crazy symbols', iirc

Happy to change this

File: src/src/pages/ecr/index.js (hunk 1 above)
Comment (andreasfrisch) [reply src/src/pages/ecr/index.js:128]: Completely forgot I had this bullshit.
Do you have any suggestions? If not I'll make some up.

File: src/src/pages/ecr/NewRepositoryDialog.js (hunk 2 above)
Comment (andreasfrisch) [reply src/src/pages/ecr/NewRepositoryDialog.js:107]: Changed the placeholder for now

File: src/src/pages/ecr/NewRepositoryDialog.js (hunk 3 above)
Comment (andreasfrisch) [reply src/src/pages/ecr/NewRepositoryDialog.js:64]: It should. Not best practices though, but I am not sure we want to or need to enforce this at this point.

**Review by kralle333 — APPROVED** [review]
Nice!

**Pull request merged**